datanode/gateway/graphql/oracles_resolver.go

```diff
@@ -67,25 +69,112 @@ func (o *oracleSpecResolver) DataConnection(ctx context.Context, obj *vegapb.Ora
 type oracleDataResolver VegaResolverRoot
 
 func (o *oracleDataResolver) ExternalData(ctx context.Context, obj *vegapb.OracleData) (*ExternalData, error) {
+	ed := &ExternalData{
+		Data: &Data{},
+	}
+
 	if obj != nil {
-		var signers []*Signer
-		if len(obj.ExternalData.Data.Signers) > 0 {
-			signers = make([]*Signer, len(obj.ExternalData.Data.Signers))
-			for i, signer := range obj.ExternalData.Data.Signers {
-				signers[i] = &Signer{
-					Signer: signer.GetSigner().(SignerKind),
+		if obj.ExternalData != nil && obj.ExternalData.Data != nil {
+			o := obj.ExternalData.Data
+			broadcastAt := strconv.FormatInt(o.BroadcastAt, 10)
+			matchedSpecs := []string{}
+			if len(o.MatchedSpecIds) > 0 {
+				matchedSpecs = o.MatchedSpecIds
+			}
+
+			properties := []*v1.Property{}
+			if len(o.Data) > 0 {
+				properties = o.Data
+			}
+
+			signers, err := resolveSigners(o.Signers)
+			if err != nil {
+				return nil, err
+			}
+
+			ed.Data = &Data{
+				Signers:        signers,
+				Data:           properties,
+				MatchedSpecIds: matchedSpecs,
+				BroadcastAt:    broadcastAt,
+			}
+		}
+	}
+
+	return ed, nil
+}
+
+func resolveSigners(obj []*v1.Signer) ([]*Signer, error) {
+	signers := []*Signer{}
+	if len(obj) > 0 {
+		for i := range obj {
+			o, signer := obj[i], &Signer{}
+
+			if pk := o.GetPubKey(); pk != nil {
+				signer.Signer = &PubKey{
+					Key: &pk.Key,
+				}
+			} else if ethAddr := o.GetEthAddress(); ethAddr != nil {
+				signer.Signer = &ETHAddress{
+					Address: &ethAddr.Address,
 				}
+			} else {
+				return nil, errors.New("invalid signer type")
```

Comment:
`o.Signer` could be `nil`, so maybe just leave `signer` as is, since we don't know if the type is wrong or it's just `nil`

Reply:
Yes, corrected this as well
Thanks!